### PR TITLE
Final changes for glvis-4.2

### PR DESCRIPTION
**Pull request by tzanio**
This will be the last PR to be merged before the glvis-4.2 release.

**Release date** 

~April 21st, 2022 (Thursday)~
May 23th, 2022 (Monday)

**TODO**

- [x] Merge https://github.com/mfem/github-actions/pull/6
- [x] Merge needs https://github.com/mfem/mfem/pull/2973
- [x] Merge #230
- [x] Merge #229 

**After the release**
- [ ] Update website, https://github.com/GLVis/web
- [ ] Update `glvis-js` and glvis.org/live
- [ ] Update `pyglvis`

**Postponed**
- Merge https://github.com/GLVis/glvis/pull/205
- Addresshttps://github.com/GLVis/glvis-js/issues/15

### CODE REVIEW

**Comment by termi-official** [comment]
Hi @tzanio , thanks for the consideration, but I think 1d elements needs more time until it is really ready to merge. The PR is somewhat functional at this point, but I think something went wrong in the way how I implemented the visualization.

**Comment by tzanio** [comment]
I added the Markdown version of the README in this branch, see https://github.com/GLVis/glvis/blob/glvis-4.2-dev/README.md. Let me know if you have any suggestions for improvements.

ping: @v-dobrev and @pazner

**Comment by v-dobrev** [comment]
> I added the Markdown version of the README in this branch, see https://github.com/GLVis/glvis/blob/glvis-4.2-dev/README.md. Let me know if you have any suggestions for improvements.
> 
> ping: @v-dobrev and @pazner

It may be helpful to add a reference to the keys/mouse functions at the beginning (maybe highlighted to draw attention) -- many people may not scroll down to see the keys/mouse sections. A good place is probably just after "For building instructions, see INSTALL."

**Comment by tzanio** [comment]
> It may be helpful to add a reference to the keys/mouse functions at the beginning (maybe highlighted to draw attention) -- many people may not scroll down to see the keys/mouse sections. A good place is probably just after "For building instructions, see INSTALL."

How is this: https://github.com/GLVis/glvis/blob/glvis-4.2-dev/README.md

**Comment by tzanio** [comment]
This is now a release candidate for `glvis-4.2`, please review and speak up if you'd like to see any changes.

**Comment by tzanio** [comment]
Can I get some reviews/approvals for this please?

File: CONTRIBUTING.md
Comment (v-dobrev) [inline CONTRIBUTING.md:294]: Why remove these?

Comment (tzanio) [reply CONTRIBUTING.md:294]: There is no version actually in `makefile` or `CMakeLists.txt`

Comment (v-dobrev) [reply CONTRIBUTING.md:294]: And `CHANGELOG`?

Comment (tzanio) [reply CONTRIBUTING.md:294]: I thought that was obvious, but I can bring that one back if you prefer

Comment (v-dobrev) [reply CONTRIBUTING.md:294]: It's better to have this list -- I see version numbers in other places that we should add: e.g. `share/Info.cmake.plist.in` and `share/Info.plist`. There maybe other places too.

Comment (tzanio) [reply CONTRIBUTING.md:294]: ok

File: CHANGELOG
Comment (v-dobrev) [inline CHANGELOG:12]: Update date?

File: CONTRIBUTING.md
Comment (v-dobrev) [reply CONTRIBUTING.md:294]: Another file that has version: `vcpkg.json`.

**Review by v-dobrev — APPROVED** [review]
A few minor issues as mentioned above.

Other than that, looks good!

File: CHANGELOG
Comment (tzanio) [reply CHANGELOG]: Done in https://github.com/GLVis/glvis/pull/227/commits/17e2a3cd6ea843ed9d7bd8213c63ce16d942345d

File: CONTRIBUTING.md
Comment (tzanio) [reply CONTRIBUTING.md]: Done in https://github.com/GLVis/glvis/pull/227/commits/17e2a3cd6ea843ed9d7bd8213c63ce16d942345d

File: README.md
Comment (kanye-quest) [inline README.md:11]: Now that we have a Markdown file, we could consider adding some repo badges here (https://github.com/badges/shields).

**Review by kanye-quest — APPROVED** [review]
LGTM

File: README.md
Comment (tzanio) [reply README.md:11]: Something like this:

                           GLVis visualization tool

                     _/_/_/  _/      _/      _/  _/
                  _/        _/      _/      _/        _/_/_/
                 _/  _/_/  _/      _/      _/  _/  _/_/
                _/    _/  _/        _/  _/    _/      _/_/
                 _/_/_/  _/_/_/_/    _/      _/  _/_/_/

                             https://glvis.org

<a href="https://github.com/GLVis/glvis/releases/latest">Release</a>
<a href="https://github.com/GLVis/glvis/actions/workflows/builds.yml">Build</a>
<a href="https://github.com/glvis/glvis/blob/master/LICENSE">License</a>
<a href="https://glvis.github.io/doxygen/html/index.html">Doxygen</a>
<a href="https://glvis.github.io/releases/glvis-macOS.dmg">License</a>
<a href="https://glvis.github.io/releases/glvis-Windows.zip">License</a>

**Pull request merged**
